### PR TITLE
DriveInfo.xml: improve for non-Windows platforms

### PR DESCRIPTION
**Pull request by alxu-avpt**

### CODE REVIEW

File: xml/System.IO/DriveInfo.xml [hunk 1]
```diff
@@ -117,22 +117,20 @@
         <Parameter Name="driveName" Type="System.String" />
       </Parameters>
       <Docs>
-        <param name="driveName">A valid drive path or drive letter. This can be either uppercase or lowercase, 'a' to 'z'. A null value is not valid.</param>
+        <param name="driveName">A valid drive letter or fully qualified path.</param>
         <summary>Provides access to information on the specified drive.</summary>
```
Comment (gewarren) [inline xml/System.IO/DriveInfo.xml:121]:
```suggestion
        <summary>Creates a new instance of the <see cref="T:System.IO.DriveInfo" /> class.</summary>
```

File: xml/System.IO/DriveInfo.xml [hunk 2]
```diff
@@ -117,22 +117,20 @@
...
         <remarks>
           <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
- Use this class to obtain information on drives. The drive name must be either an uppercase or lowercase letter from 'a' to 'z'. You cannot use this method to obtain information on drive names that are `null` or use UNC (\\\server\share) paths.  
+ Use this class to obtain information on drives. You cannot use this method to obtain information on drive names that are `null` or use UNC (\\\server\share) paths.
```
Comment (gewarren) [inline xml/System.IO/DriveInfo.xml:126]:
```suggestion
You can't use this constructor to obtain information on drive names that are <see langword="null" /> or use UNC (`\\server\share`) paths.  
```

File: xml/System.IO/DriveInfo.xml [hunk 3]
```diff
@@ -117,22 +117,20 @@
...
+        <param name="driveName">A valid drive letter or fully qualified path.</param>
         <summary>Provides access to information on the specified drive.</summary>
         <remarks>
           <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
- Use this class to obtain information on drives. The drive name must be either an uppercase or lowercase letter from 'a' to 'z'. You cannot use this method to obtain information on drive names that are `null` or use UNC (\\\server\share) paths.  
+ Use this class to obtain information on drives. You cannot use this method to obtain information on drive names that are `null` or use UNC (\\\server\share) paths.  
+
+ On Windows, passing a mounted directory (e.g. C:\NetworkMountedDrive) returns the root drive information (e.g. C:).
```
Comment (gewarren) [inline xml/System.IO/DriveInfo.xml:128]: The constructor doesn't technically return anything, so this should be reworded.

**Review by gewarren — CHANGES_REQUESTED** [review]

**Review by adamsitnik — COMMENTED** [review]
@alxu-avpt could you please address the feedback from @gewarren ?

**Comment by alxu-avpt** [comment]
@dotnet-policy-service agree company="AvePoint"

**Comment by alxu-avpt** [comment]
> @alxu-avpt could you please address the feedback from @gewarren ?

Sorry, I was applying for CLA approval from my company. Please re-review, thanks!

**Review by gewarren — APPROVED** [review]
Thanks @alxu-avpt

File: xml/System.IO/DriveInfo.xml [hunk 4]
```diff
@@ -117,22 +117,20 @@
...
-        <param name="driveName">A valid drive path or drive letter. This can be either uppercase or lowercase, 'a' to 'z'. A null value is not valid.</param>
-        <summary>Provides access to information on the specified drive.</summary>
+        <param name="driveName">A valid drive letter or fully qualified path.</param>
+        <summary>Creates a new instance of the <see cref="T:System.IO.DriveInfo" /> class.</summary>
         <remarks>
           <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
- Use this class to obtain information on drives. The drive name must be either an uppercase or lowercase letter from 'a' to 'z'. You cannot use this method to obtain information on drive names that are `null` or use UNC (\\\server\share) paths.  
+ You can't use this constructor to obtain information on drive names that are <see langword="null" /> or use UNC (`\\server\share`) paths.
```
Comment (gewarren) [inline xml/System.IO/DriveInfo.xml:126]:
```suggestion
 You can't use this constructor to obtain information on drive names that are `null` or use UNC (`\\server\share`) paths.
```

**Pull request merged**